src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs

```diff
@@ -0,0 +1,275 @@
+            /// This throws if the column is not active in this row, or if the type
+            /// <typeparamref name="TValue"/> differs from this column's type.
+            /// </summary>
+            /// <typeparam name="TValue"> is the column's content type.</typeparam>
+            /// <param name="column"> is the output column whose getter should be returned.</param>
+            public override ValueGetter<TValue> GetGetter<TValue>(DataViewSchema.Column column)
+            {
+                Ch.CheckParam(column.Index < _getters.Length, nameof(column), "requested column not valid.");
+                Ch.Check(IsColumnActive(column));
+
+                var fn = _getters[column.Index] as ValueGetter<TValue>;
+                if (fn == null)
+                    throw Ch.Except("Invalid TValue in GetGetter: '{0}'", typeof(TValue));
+                return fn;
+            }
+
+            private Delegate CreateGetterDelegate(int col)
+            {
+                return Utils.MarshalInvoke(CreateGetterDelegate<int>, _bindings.Infos[col].ColType.RawType, col);
+            }
+
+            private Delegate CreateGetterDelegate<TValue>(int col)
+            {
+                Delegate getterDelegate;
+
+                var colInfo = _bindings.Infos[col];
+                Ch.Assert(colInfo.ColType.RawType == typeof(TValue));
+
+                if (typeof(TValue) == typeof(bool))
+                {
+                    getterDelegate = CreateBooleanGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(byte))
+                {
+                    getterDelegate = CreateByteGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(DateTime))
+                {
+                    getterDelegate = CreateDateTimeGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(double))
+                {
+                    getterDelegate = CreateDoubleGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(short))
+                {
+                    getterDelegate = CreateInt16GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(int))
+                {
+                    getterDelegate = CreateInt32GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(long))
+                {
+                    getterDelegate = CreateInt64GetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(sbyte))
+                {
+                    getterDelegate = CreateSByteGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(float))
+                {
+                    getterDelegate = CreateSingleGetterDelegate(colInfo);
+                }
+                else if (typeof(TValue) == typeof(string))
```

Comment:
For strings, this type is usually `typeof(ReadOnlyMemory<char>)`. ML.NET uses `ROM<char>` everywhere in order to support sub-strings and not having to allocate each individual string.

We should be sure to add a test for strings to cover this.

Reply:
`DbDataReader` does not support `Span` or `ReadOnlyMemory`.

Reply:
Correct, but ML.NET's IDataView only supports `ReadOnlyMemory<char>` and not `string`. So `TValue` will be of type `ReadOnlyMemory<char>`, and you will need to set the `ref` to the result from `DbDataReader.GetString()`.

See https://github.com/dotnet/machinelearning/blob/45b6c43387294d19cad843a21e87539213e8d34c/src/Microsoft.ML.DataView/DataViewType.cs#L84-L87

When `IDataView` inspects string values, it uses `ReadOnlyMemory<char>`. So you will need to turn the `string` you get from DbDataReader into a `ReadOnlyMemory<char>` here.

Reply:
Fixed.